File: rb/BeginnersSelection/03_ShiftOnly.rb
# https://atcoder.jp/contests/abs/tasks/abc081_b

class Object
  def let(f)
    f.call self
  end
end

[gets, gets][1]
  .chomp
  .split(" ")
  .map { |x| x.to_i.to_s(2) }
  .map { |x| x.length - x.rindex('1') - 1 }
  .min
  .let(method(:puts))

# https://atcoder.jp/contests/abs/submissions/21508284

File: py/BeginnersSelection/03_ShiftOnly.py
# https://atcoder.jp/contests/abs/tasks/abc081_b

class W:
  def __init__(self, v): self.v = v
  let = lambda self, f: W(f(self.v))
  map = lambda self, f: W(map(f, self.v))

W([input(), input()][1].split()) \
  .map(int) \
  .map(bin) \
  .map(lambda x: len(x) - x.rindex("1") - 1) \
  .let(min) \
  .let(print)

# https://atcoder.jp/contests/abs/submissions/21596351
